plugins/module_utils/network/iosxr/facts/ospf_interfaces/ospf_interfaces.py

```diff
@@ -0,0 +1,135 @@
+based on the configuration.
+"""
+
+from copy import deepcopy
+import re
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common import (
+    utils,
+)
+from ansible_collections.cisco.iosxr.plugins.module_utils.network.iosxr.rm_templates.ospf_interfaces import (
+    Ospf_interfacesTemplate,
+)
+from ansible_collections.cisco.iosxr.plugins.module_utils.network.iosxr.argspec.ospf_interfaces.ospf_interfaces import (
+    Ospf_interfacesArgs,
+)
+
+
+class Ospf_interfacesFacts(object):
+    """ The iosxr ospf_interfaces facts class
+    """
+
+    def __init__(self, module, subspec="config", options="options"):
+        self._module = module
+        self.argument_spec = Ospf_interfacesArgs.argument_spec
+        spec = deepcopy(self.argument_spec)
+        if subspec:
+            if options:
+                facts_argument_spec = spec[subspec][options]
+            else:
+                facts_argument_spec = spec[subspec]
+        else:
+            facts_argument_spec = spec
+
+        self.generated_spec = utils.generate_dict(facts_argument_spec)
+
+    def get_ospf_interfaces(self, connection, flag):
+        cmd = "show running-config router " + flag
+        return connection.get(cmd)
+
+    def populate_facts(self, connection, ansible_facts, data=None):
+        """ Populate the facts for Ospf_interfaces network resource
+
+        :param connection: the device connection
+        :param ansible_facts: Facts dictionary
+        :param data: previously collected conf
+
+        :rtype: dictionary
+        :returns: facts
+        """
+        facts = {}
+        objs = []
+        if not data:
+            data = self.get_ospf_interfaces(connection, flag="ospf")
+            data += "\n" + self.get_ospf_interfaces(connection, flag="ospfv3")
+        end_flag, end_mark, count, v_read = 0, 0, 0, False
+        areas, config_commands = [], []
+        area_str, process, curr_process = "", "", ""
+        data = data.splitlines()
+
+        for line in data:
+            if (
+                line.startswith("router")
+                and curr_process != ""
+                and curr_process != line
+            ):
+                end_mark, count, end_flag, area_str = 0, 0, 0, ""
+            if end_mark == 0 and count == 0 and line.startswith("router ospf"):
+                curr_process = line
+                process = re.sub("\n", "", line)
+                count += 1
+                config_commands.append(process)
+            else:
+                if line.startswith(" area") or line.startswith(" vrf"):
+                    area_str = process + re.sub("\n", "", line)
+                    config_commands.append(area_str.replace("  ", " "))
+                    end_flag += 1
+                elif line.startswith("  interface"):
+                    virtual_str = area_str + re.sub("\n", "", line)
+                    config_commands.append(virtual_str.replace("  ", " "))
```

Comment:
replace code is used often, can u plz comment with example scenario for future maintenance